src/combinatorics/quiver/basic.lean

```diff
@@ -124,4 +124,7 @@ instance empty_quiver (V : Type u) : quiver.{u} (empty V) := ⟨λ a b, pempty
 
 @[simp] lemma empty_arrow {V : Type u} (a b : empty V) : (a ⟶ b) = pempty := rfl
 
+/-- A quiver is thin if it has no parallel arrows. -/
+def is_thin (V : Type u) [quiver V] := ∀ (a b : V), subsingleton (a ⟶ b)
```

Comment:
Is there any point in having this as a definition (as opposed to an abbreviation, os just spelling it out)? For reference, this is to `subsingleton` what `decidable_rel` is to `decidable`. Also note that this is already used sppelled out in the `category_theory.skeletal` file and around.

Reply:
Well, the definition originally was made for groupoids in order to use it for subgroupoids, and then moved here since it actually only has to do with quivers. I don't mind an abbreviation, or even getting rid of it entirely, though I would have guessed it would probably important enough elsewhere too to deserve a definition.

Reply:
I think this should be an abbreviation and treated as an instance.

Reply:
OK, well, I pushed a version with `is_thin` dropped from quivers altogether. I guess I should have waited for a consensus first. I'll abstain from adding the definition back until so, is that alright?

In any case, I think I would agree with keeping the definition/abbreviation, and making use of it, e.g. also adding it explicitly to the `skeletal` code.

Reply:
Since `abbreviation` is reducible, the instance thingy can see through it, and `[is_thin V]` is the same as `[∀ (a b : V), subsingleton (a ⟶ b)]` when used as an instance argument.

Reply:
TIL, thanks!

Reply:
`@[reducible] def` is usually preferred over `abbreviation` thought, see [Zulip](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/.40.5Breducible.5D.20defs.20of.20classes.20on.20the.20leadup.20to.20Lean4.2E/near/292160315).